flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/NettyShuffleMetricFactory.java

```diff
@@ -154,24 +157,40 @@ private static void registerOutputMetrics(
 
 	public static void registerInputMetrics(
 			boolean isDetailedMetrics,
+			boolean isCreditBased,
 			MetricGroup inputGroup,
 			SingleInputGate[] inputGates) {
 		registerInputMetrics(
 			isDetailedMetrics,
+			isCreditBased,
 			inputGroup,
 			inputGroup.addGroup(METRIC_GROUP_BUFFERS),
 			inputGates);
 	}
 
 	private static void registerInputMetrics(
 			boolean isDetailedMetrics,
+			boolean isCreditBased,
 			MetricGroup inputGroup,
 			MetricGroup buffersGroup,
 			SingleInputGate[] inputGates) {
 		if (isDetailedMetrics) {
 			InputGateMetrics.registerQueueLengthMetrics(inputGroup, inputGates);
 		}
+
 		buffersGroup.gauge(METRIC_INPUT_QUEUE_LENGTH, new InputBuffersGauge(inputGates));
+
+		if (isCreditBased) {
+			FloatingBuffersUsageGauge floatingBuffersUsageGauge = new FloatingBuffersUsageGauge(inputGates);
+			ExclusiveBuffersUsageGauge exclusiveBuffersUsageGauge = new ExclusiveBuffersUsageGauge(inputGates);
+
+			buffersGroup.gauge(METRIC_INPUT_EXCLUSIVE_BUFFERS_USAGE, exclusiveBuffersUsageGauge);
+			buffersGroup.gauge(METRIC_INPUT_FLOATING_BUFFERS_USAGE, floatingBuffersUsageGauge);
+			buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new CreditBasedInputBuffersUsageGauge(floatingBuffersUsageGauge,
```

Comment:
It is not good to make partial parameters in separate line. I think you should make `METRIC_INPUT_POOL_USAGE` as a separate line, and `new CreditBasedInputBuffersUsageGauge(floatingBuffersUsageGauge, exclusiveBuffersUsageGauge, inputGates)` as a separate line.
Or you could define a local var for `CreditBasedInputBuffersUsageGauge` like above `floatingBuffersUsageGauge` and `exclusiveBuffersUsageGauge`.

Reply:
nit: I think it is better to define a local var for `CreditBasedInputBuffersUsageGauge` like above `floatingBuffersUsageGauge` and `exclusiveBuffersUsageGauge`, because the broken line seems not good here.

Reply:
ok

Reply:
@zhijiangW do you have other suggestions, I can solve them together today to merge this before 1.9 release

Reply:
I think I have issues only for the tests, and I would post then later today.

Reply:
ok, very thanks